docs/testplan/smart-switch/high-availability/keysight_ha_testplan.md

```diff
@@ -0,0 +1,259 @@
+additionally be used with Stateful traffic for VxLan encapsulation and decapsulation.
+
+![HA SmartSwitch Topology](images/smartswitch_ha_topology.svg)
+
+## Topology
+### Configuration for HA testing
+The topic presented is an interim test plan for SmartSwitch HA testing.  The primary goal is to
+illustrate a vision for simulating multiple HA scenarios using a SmartSwitch mock-up to test DPU
+SmartSwitch behavior.   This document is based on testing now before full Sonic integration.
+
+![HA topology](images/ha_test_topology.svg)
+
+## Setup Configuration
+Two DPUs will use similar network configurations (VxLAN, ENI) forming an HA set.  One DPU will be
+designated as the Active the second Standby.
+
+## Test Methodology
+Following test methodology will be used for measuring HA switchover and performance.
+* Traffic generator will be used to configure ENI peering between DPU ports.
+* Data traffic will be sent from  server to server, server to SmartSwitch and SmartSwitch to server.
+* Depending on the test case, switchovers/failovers will be generated and measured.
+* Switchover between DPUs will be measured by noting down the precise time of the
+switchover/failover event.  Traffic generator will create those timestamps and provide us with
+the recovery statistics.
+
+
+## Test cases
+
+### Test Metrics
+Each test case will use the following metrics to measure performance of the HA
+failover/switchover for all test cases.
+
+The following metrics shall be collected Before and After switchover is initiated.
+
+| Performance Metric                  | Achieved |
+|:------------------------------------|:--------:|
+| Concurrent Connections              |         |
+| CPS                                 |         |
+| PPS                                 |         |
+| TCP Client Failures Resets Sent     |         |
+| TCP Client Failures Resets Recieved |         |
+| TCP Client Failures Resets Retries  |         |
+| TCP Server Failures Resets Sent     |         |
+| TCP Server Failures Resets Recieved |         |
+| TCP Server Failures Resets Retries  |         |
+
+As soon as failover/switchover is initiated a timer will be initiated to measure DPU pair switchover.
+
+| Failover/Switchover<br/>Type |                   Event                    | Convergence (s) |
+|:------------------------|:------------------------------------------:| :---: |
+|                         | Switchover time from <br/>Active-to-Standby DPU | 0  |
+
+### Test case1 CPS Without HA Enabled
+#### Test Objective 1
+* Baseline performance of the SmartSwitch system before HA sets are established.
+* Test metrics will be collected on a single DPU within a SmartSwitch
+
+![HA CPS Without HA Enabled](images/ha_test_cps_ha_not_enabled.svg)
+#### Steps for Test Case 1
+* Configure 1 DPUs networked within a SmartSwitch with HA not enable.
+* The SmartSwitch configuration will consist of 1 DPU sharing the same network configurations
+such as: IPs, MACs, VLan/VxLan, ENIs.
```

Comment:
Smart switch will have a different configuration than DPU, such as MAC address and IP. When smart switch receives traffic, it forwards it to the DPU.